_posts/2020-01-13-gemifying-your-styleguide.markdown

```diff
@@ -0,0 +1,113 @@
+
+In this article I'll show you how we made it!
+
+<!--more-->
+
+## Creating the Gem
+
+The first step is to create the gem. Bundler has a command to create the skeleton of the Gem.
+
+  `bundle gem ombulabs-styleguide`
+
+Go to the created `.gemspec` file and add the info about the new gem and also a few dependencies:
+
+```ruby
+# ombulabs-styleguide.gemspec
+require_relative 'ombulabs/styleguide/version'
+
+Gem::Specification.new do |spec|
+  spec.name           = 'ombulabs-styleguide'
+  spec.version        = Styleguide.gem_version
+  spec.authors        = ['OmbuLabs']
+  spec.email          = ['hello@ombulabs.com']
+
+  spec.summary        = 'Style Guide for all OmbuLabs products'
+  spec.homepage       = 'https://github.com/ombulabs/styleguide'
+
+  # Rails
+  spec.add_dependency 'rails', '>= 5.2.1'
+  spec.add_dependency 'sass-rails', '>= 5.0'
+  # Jquery
+  spec.add_dependency 'jquery-rails', '>= 4.3.0'
+  # Bootstrap
+  spec.add_dependency 'bootstrap-sass', '>= 3.4.0'
+  # Popper
+  spec.add_dependency 'popper_js', '>= 1.14.5'
+  spec.add_dependency 'material_design_lite-sass', '>= 1.3.0'
+end
+```
+
+By default the gem is created as a module, but we need to turn into an engine, because we need to be able to integrate the gem code into any Rails application. In case you want to understand more about Rails engine, check [this link](https://guides.rubyonrails.org/engines.html).
+
+So, let's change the `lib/ombulabs-styleguide.rb`
+
+```ruby
+module Ombulabs
+  module Styleguide
+    class Engine < ::Rails::Engine
+    end
+  end
+end
+```
+
+Now we need to copy all the assets for the `vendor\assets` folder. The project structure will be like this:
+
+<img src="/blog/assets/images/gemifying-your-styleguide-img-01.png" alt="Folder structure" style="width: 300px; margin: 0;">
+
+Go back to the .gemspec file and make sure that the new files will be loaded adding the following line:
+
+```ruby
+spec.files = Dir['vendor/**/*']
+```
+
+And this is it! The gem is ready to be used.
+
+## Adding the gem to a project
+
+To use our new created gem we need to bundle it like any other gem. So add this line to your application's Gemfile:
+
+```ruby
+gem 'ombulabs-styleguide'
+```
+
+And then execute:
+
+`$ bundle install`
+
+The next step is to require the new gem assets into the application:
+
+If using SASS, in `application.scss`, add:
+
+```sass
+@import "ombulabs/styleguide";
+```
+
+And in `application.js` add:
+
+```javascript
+//= require ombulabs/styleguide
+```
+
+Now start the application and the gem assets should be available to use.
+
+You can see our style guide code in this GitHub repository: [https://github.com/fastruby/styleguide](https://github.com/fastruby/styleguide).
```

Comment:
@cleicar shouldn't you link the one for ombulabs here? since that's the one you talked in the article

Reply:
@lubc yeah this is a bit confuse because the new Ombu Labs styleguide is hosted inside Fast Ruby account, like we can see here: https://github.com/fastruby/styleguide. It says "Ombulabs::Styleguide". 

But the hosted page says that it's "FastRuby Style Guide": https://fastruby.github.io/styleguide

So there a few things we need to know:
- Will this style guide be applied to Ombulabs products or only Fast Ruby's?
- Should we fix Fast Ruby styleguide to say that is Ombu Labs styleguide?
- Should we update Ombu Labs styleguide before release this article?

@ABizzinotto @etagwerker I would like to know your thoughts.

Reply:
> the new Ombu Labs styleguide is hosted inside Fast Ruby account, like we can see here: https://github.com/fastruby/styleguide.

The OmbuLabs style guide is here: 

- https://github.com/ombulabs/styleguide
- https://ombulabs.github.io/styleguide/

>  It says "Ombulabs::Styleguide".

This is a typo in the FastRuby.io style guide. It should be corrected in the README: https://github.com/fastruby/styleguide

Reply:
Thanks @etagwerker.

So this article should be about Fast Ruby style guide (since this is the one we turned into a gem). 

I suggested move this article to Fast Ruby blog and I'll do a few changes. What do you think?